src/jit/compiler.cpp

```diff
@@ -1007,6 +1008,7 @@ var_types Compiler::getReturnTypeForStruct(CORINFO_CLASS_HANDLE clsHnd,
         // and also examine the clsHnd to see if it is an HFA of count one
         useType = getPrimitiveTypeForStruct(structSize, clsHnd);
     }
+#endif
```

Comment:
#10027 states that Identity_HFA01 return by EAX (without RetBuf). Does this change reflect that? It seems that we need to use ``structSize <= sizeof(float)`` instead of ``structSize <= sizeof(double)``.